Sources/SwiftKafka/KafkaConfig.swift

```diff
@@ -75,9 +75,28 @@ public struct KafkaConfig: Hashable, Equatable {
             }
         }
 
+        func setDeliveryReportCallback(
+            callback: @escaping (@convention(c) (OpaquePointer?, UnsafePointer<rd_kafka_message_t>?, UnsafeMutableRawPointer?) -> Void)
+        ) {
+            rd_kafka_conf_set_dr_msg_cb(
+                self.pointer,
+                callback
+            )
+        }
+
+        func setOpaque(opaque: AnyObject) {
+            rd_kafka_conf_set_opaque(
+                self.pointer,
+                Unmanaged.passUnretained(opaque).toOpaque()
```

Comment:
I find this more confusing than before that the `setOpaque` method is now hiding the `passUnretained` from the caller. I think @Lukasa meant something slightly different here where you are going to have an object that handles the `passUnretained`/`takeUnretained` for you and you just have to keep a reference to that object.

However, @Lukasa can probably expand more on this.

Reply:
Ahh ok, I think I understand it now, please check in my latest commit!